app/org/sagebionetworks/bridge/services/SurveyService.java

```diff
@@ -291,6 +295,20 @@ public Survey getSurveyMostRecentlyPublishedVersion(StudyIdentifier studyIdentif
 
         return surveyDao.getAllSurveysMostRecentVersion(studyIdentifier, includeDeleted);
     }
+
+    /**
+     * Callers must operate on a survey in their own study. However our code has allowed administrators to delete 
+     * shared studies (which are not in the admin's study). For backwards compatibility, do not enforce the same 
+     * study rule for shared study surveys. Eventually we want admins to be able to switch into the shared study in 
+     * order to delete items there, then this exception to the check can be removed. 
+     */
+    private boolean isInStudy(StudyIdentifier studyId, Survey survey) {
+        if (studyId == null || survey == null || survey.getStudyIdentifier() == null) {
+            return true;
+        }
+        return survey.getStudyIdentifier().equals("shared") || 
+               survey.getStudyIdentifier().equals(studyId.getIdentifier());
```

Comment:
How should we secure shared modules? Should they only be accessible to admins (presumably all Sage employees)? If so I will check and throw an exception if this is not true in the controller.

Reply:
I believe if we follow the model that only shared project devs/admins can change that study, it is secure, so I'll remove this exception for "shared" modules. The integration test will have to find another way to delete test modules (more likely, check and create a test module if it's not there, rather than creating/deleting each time.

Reply:
I've replaced this with passing the user's roles into the service for a couple of methods, and skipping the study comparison check for admins. This is temporary until we have an API for admins to adopt any study